recipes/HeterogenousArrayLog/README.md

```diff
@@ -0,0 +1,16 @@
+# HeterogenousArrayLog
+
+This recipe demonstrates how to create a heterogenous array and process its elements.
+
+PureScript arrays are _homogeneous_, meaning that all values must have the same type. If you want to store values with different types in the same array (i.e. a _heterogeneous_ array), you should wrap all the types you wish to store in either a [sum type](https://github.com/purescript/documentation/blob/master/language/Types.md#tagged-unions) or a [Variant](https://pursuit.purescript.org/packages/purescript-variant). This recipe demonstrates both strategies.
+
```

Comment:
A note about pros and cons of either approach may be helpful. I'm not aware of any besides personal preference for style.

Reply:
Variant is extensible, so it comes with the pros/cons of that. However, anything row-related will likely incur a performance cost in some situations due to type class dictionaries.